stack-graphs/src/partial.rs

```diff
@@ -2857,16 +2873,28 @@ impl Path {
         let mut lhs_scope_stack_postcondition = self.scope_stack;
         let mut rhs_symbol_stack_precondition = partial_path.symbol_stack_precondition;
         let mut rhs_scope_stack_precondition = partial_path.scope_stack_precondition;
-        graph[self.end_node].halfopen_closed_postcondition(
+        if let Err(e) = graph[self.end_node].halfopen_closed_postcondition(
             paths,
             &mut lhs_symbol_stack_postcondition,
             &mut lhs_scope_stack_postcondition,
-        );
-        graph[partial_path.start_node].halfopen_closed_partial_precondition(
+        ) {
+            panic!(
+                "failed to halfopen postcondition of {}: {:?}",
+                self.display(graph, paths),
+                e
+            );
+        }
+        if let Err(e) = graph[partial_path.start_node].halfopen_closed_partial_precondition(
```

Comment:
Slight style nit, I think this might be clearer with `unwrap_or_else`:

``` rust
        graph[self.end_node].halfopen_closed_postcondition(
            paths,
            &mut lhs_symbol_stack_postcondition,
            &mut lhs_scope_stack_postcondition,
        ).unwrap_or_else(|| panic(/* etc */));
```

(This works because an expression that returns `()` can be used as a statement without needing `let _ =`)

Having typed it up I'm now only 50% confident I like the unwrap version better.  Your call!

Reply:
I like it too. The panic is really secondary, and this way we don't have the top-level construct (`if`) there only for the unhappy path.